igvc_navigation/src/navigation_client/navigation_client.cpp

```diff
@@ -0,0 +1,185 @@
+      double lon = (tokens[1].find('?') != std::string::npos) ? convertDmsToDec(tokens[1]) : stod(tokens[1]);
+
+      // transform latitude and longitude to UTM frame
+      geometry_msgs::PointStamped waypoint_utm;
+      RobotLocalization::NavsatConversions::UTM(lat, lon, &(waypoint_utm.point.x), &(waypoint_utm.point.y));
+      waypoint_utm.header.frame_id = "utm";
+
+      // transform utm frame to odom frame
+      geometry_msgs::PointStamped waypoint_odom;
+      tf_listener_.transformPoint("odom", ros::Time(0), waypoint_utm, "odom", waypoint_odom);
+      waypoints_list_.push_back(waypoint_odom);
+    }
+    line_index++;
+  }
+
+  ROS_INFO_STREAM_ONCE(waypoints_list_.size() << " waypoints loaded.");
+}
+
+double NavigationClient::convertDmsToDec(std::string dms)
+{
+  auto q_mark_iter = dms.find('?');
+  auto apos_iter = dms.find('\'');
+  auto quote_iter = dms.find('\"');
+  auto degrees = stod(dms.substr(0, q_mark_iter));
+  auto minutes = stod(dms.substr(q_mark_iter + 1, apos_iter));
+  auto seconds = stod(dms.substr(apos_iter + 1, quote_iter));
+  auto dir_char = dms[dms.size() - 1];
+
+  const double seconds_in_minute = 60.0;
+  const double seconds_in_hour = 3600.0;
+  degrees += minutes / seconds_in_minute;
+  degrees += seconds / seconds_in_hour;
+
+  if (dir_char == 'W' || dir_char == 'S')
+    degrees *= -1;
+
+  return degrees;
+}
+
+void NavigationClient::sendPoseAsGoal(const geometry_msgs::PoseStamped& pose)
+{
+  ROS_INFO_STREAM("Sending pose: (" << pose.pose.position.x << ", " << pose.pose.position.y
+                                    << ") with yaw = " << tf::getYaw(pose.pose.orientation));
+  mbf_msgs::MoveBaseGoal goal;
+  goal.target_pose = pose;
+
+  client.sendGoal(goal);
+}
+
+void NavigationClient::sendPoseAsGoalAndWait(const geometry_msgs::PoseStamped& pose)
+{
+  ROS_INFO_STREAM("Sending pose and waiting: (" << pose.pose.position.x << ", " << pose.pose.position.y
+                                                << ") with yaw = " << tf::getYaw(pose.pose.orientation));
+  mbf_msgs::MoveBaseGoal goal;
+  goal.target_pose = pose;
+
+  client.sendGoalAndWait(goal);
+}
+
+void NavigationClient::sendPointAsGoal(const geometry_msgs::PointStamped& point)
+{
+  ROS_INFO_STREAM("Sending point: (" << point.point.x << ", " << point.point.y << ") with yaw = 0");
+  mbf_msgs::MoveBaseGoal goal;
+
+  goal.target_pose.header = point.header;
+  goal.target_pose.pose.position = point.point;
+  goal.target_pose.pose.orientation = tf::createQuaternionMsgFromYaw(0.0);
+
+  client.sendGoal(goal);
+}
+
+void NavigationClient::sendPointAsGoalAndWait(const geometry_msgs::PointStamped& point)
+{
+  ROS_INFO_STREAM("Sending point: (" << point.point.x << ", " << point.point.y << ") with yaw = 0");
+  mbf_msgs::MoveBaseGoal goal;
+
+  goal.target_pose.header = point.header;
+  goal.target_pose.pose.position = point.point;
+  goal.target_pose.pose.orientation = tf::createQuaternionMsgFromYaw(0.0);
+
+  client.sendGoalAndWait(goal);
+}
```

Comment:
I feel like there's some code duplication going on in these four functions here. I guess you could have the "wait" part be a parameter, and then have `Point` variants be overloads of the `Pose` ones.